packages/block-editor/src/components/block-list/moving-animation.js

```diff
@@ -28,7 +29,8 @@ import { useReducedMotion } from '@wordpress/compose';
  * @return {Object} Style object.
  */
 function useMovingAnimation( ref, isSelected, enableAnimation, triggerAnimationOnChange ) {
-	const prefersReducedMotion = useReducedMotion() || ! enableAnimation;
+	const isTyping = useSelect( ( select ) => select( 'core/block-editor' ).isTyping() );
```

Comment:
Ideally, the animation stays agnostic to the store. At some point in the future we might want to move it to a generic hook. We can use the `enableAnimation` argument to pass this as an argument from the outside.

Reply:
@youknowriad That was my initial thought as well, but there's a comment at the bottom of the parent component saying that it should be limited to checking typing for the selected block to reduce re-renders. We could wrap `useMovingAnimation` when the time comes to reuse it elsewhere?